source/codegen/templates/service_helpers.mako

```diff
@@ -773,6 +774,9 @@ ${set_response_values(normal_outputs, init_method)}\
   else:
     method_call = f'return ConvertApiErrorStatusFor{cpp_handle_type}(context, status, {session});'
 %>\
+      %if error_parameter is not None:
+      context->AddTrailingMetadata("${error_parameter["return_on_error_key"]}", std::to_string(${common_helpers.pascal_to_snake(error_parameter["name"])}));
+      %endif
```

Comment:
I don't think the helper function helps very much. It duplicates the output check and constrains this to supporting one return_on_error_key parameter.

```suggestion
      %for parameter in output_parameters:
      %if "return_on_error_key" in parameter:
      context->AddTrailingMetadata("${parameter["return_on_error_key"]}", std::to_string(${common_helpers.pascal_to_snake(parameter["name"])}));
      %endif
      %endfor
```

Reply:
Removed the helper method and updated the mako as per the suggestion